web/webkit/src/main/scala/net/liftweb/http/LiftRules.scala

```diff
@@ -2295,3 +2301,80 @@ trait FormVendor {
   private object requestForms extends SessionVar[Map[String, List[FormBuilderLocator[_]]]](Map())
 }
 
+abstract class LiftRulesSettingException(settingName: String, stackTrace: LiftRules.StackTrace, message: String) extends Exception(message) {
+  setStackTrace(stackTrace)
+}
+
+case class LiftRulesSettingWrittenAfterReadException(settingName: String, stackTrace: LiftRules.StackTrace, message: String)
+  extends LiftRulesSettingException(settingName, stackTrace, message)
+
+case class LiftRulesSettingWrittenAfterBootException(settingName: String, stackTrace: LiftRules.StackTrace, message: String)
+  extends LiftRulesSettingException(settingName, stackTrace, message)
+
+case class LiftRulesSettingWrittenTwiceException(settingName: String, stackTrace: LiftRules.StackTrace, message: String)
+  extends LiftRulesSettingException(settingName, stackTrace, message)
+
+
+class LiftRulesSetting[T](val name: String, val default: T) extends LiftValue[T] with HasCalcDefaultValue[T] {
```

Comment:
Some scaladocs would be 👌, particularly advising when you would use this vs another approach.